connectors/http/polling/src/main/java/io/camunda/connector/http/polling/SchedulingService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.http.polling;
+
+import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
+import io.camunda.connector.api.inbound.operate.ProcessInstance;
+import io.camunda.connector.http.base.services.HttpService;
+import io.camunda.connector.http.polling.task.RequestPollingTask;
+import io.camunda.connector.http.polling.task.SearchNewProcessesTask;
+import io.camunda.connector.http.polling.util.DelayParser;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SchedulingService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SchedulingService.class);
+
+  private final ScheduleStore scheduleStore;
+  private final HttpService httpService;
+  private final PollingScheduler pollingScheduler;
+
+  public SchedulingService(
+      final ScheduleStore scheduleStore,
+      final HttpService httpService,
+      final PollingScheduler pollingScheduler) {
+    this.scheduleStore = scheduleStore;
+    this.httpService = httpService;
+    this.pollingScheduler = pollingScheduler;
+  }
+
+  public void schedulePollingTaskForContext(final InboundIntermediateConnectorContext context) {
+
+    ScheduledFuture<?> searchNewProcessesTask = scheduleStore.getSearchNewProcessesTask();
+    if (searchNewProcessesTask != null) {
+      boolean cancel = searchNewProcessesTask.cancel(true);
+      LOGGER.debug("Cancelled existing search task: {}", cancel);
+    }
+    ScheduledFuture<?> scheduledFuture =
+        pollingScheduler.scheduleTask(
+            new SearchNewProcessesTask(scheduleStore, getScheduleNewInstanceTask()),
+            DelayParser.getInitialDelay(context),
+            DelayParser.getPollingInterval(context),
+            TimeUnit.MILLISECONDS);
+    scheduleStore.updateSearchNewProcessesTask(context, scheduledFuture);
+  }
+
+  private BiFunction<InboundIntermediateConnectorContext, ProcessInstance, ScheduledFuture<?>>
+      getScheduleNewInstanceTask() {
+    return (context, newProcessInstance) ->
+        pollingScheduler.scheduleTask(
+            new RequestPollingTask(
+                context,
+                newProcessInstance,
+                httpService,
+                getCancelTaskFunction(),
+                new ConcurrentHashMap<>()),
+            0,
+            15,
+            TimeUnit.SECONDS);
```

Comment:
Why do we use the same polling scheduler here? 

I recommend to have two components:
- ProcessInstanceManager (this component is polling all the process definitions and their active process instances for all the inbound polling connectors)
- PollingJobScheduler (this scheduler only manages the actual next polling requests) as this is a separate concern.